src/Featurizers/NormalizeFeaturizer.h

```diff
@@ -0,0 +1,417 @@
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         NormalizeTransformer
+///  \brief         Normalize based on l1, l2 or max norm
+///
+template <typename IteratorRangeT>
+class NormalizeTransformer : public StandardTransformer<IteratorRangeT, std::vector<std::double_t>> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+
+    // checks if IteratorRangeT is a pair of iterators
+    static_assert(Components::IsIteratorPair<IteratorRangeT>::value, "VectorNormsEstimator input type need to a pair of iterators of same type!");
+
+
+    using BaseType     = StandardTransformer<IteratorRangeT, std::vector<std::double_t>>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    NormalizeTransformer(std::vector<std::double_t> norms);
+    NormalizeTransformer(Archive &ar);
+
+    ~NormalizeTransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(NormalizeTransformer);
+
+    bool operator==(NormalizeTransformer const &other) const;
+
+    void save(Archive &ar) const override;
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Types
+    // |
+    // ----------------------------------------------------------------------
+    using IteratorType = typename std::tuple_element<0, IteratorRangeT>::type;
+    using ValueType    = typename std::iterator_traits<IteratorType>::value_type;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    std::vector<std::double_t> const                   _norms;
+    size_t                                             _row;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+    void execute_impl(IteratorRangeT const &input, typename BaseType::CallbackFunction const &callback) override;
+
+    void execute_impl(IteratorType &begin, IteratorType const &end, typename BaseType::CallbackFunction const &callback, std::true_type);
+    void execute_impl(IteratorType &begin, IteratorType const &end, typename BaseType::CallbackFunction const &callback, std::false_type);
+
+    bool is_null(IteratorType const &input, std::true_type);
+    bool is_null(IteratorType const &input, std::false_type);
+
+    typename GetNullableType<ValueType>::type get_value(IteratorType const &input, std::true_type);
+    ValueType get_value(IteratorType const &input, std::false_type);

```

Comment:
These functions aren't mutating data, so they should be `const`, however...
These functions aren't using any class data, so they should be `static`. however...
These functions aren't adding functionality to the class, so they should be free functions, probably declared in an anonymous namespace.